### PR TITLE
Paul/global search view 0.1

### PR DESCRIPTION
**Pull request by paukparl**
- SearchSubscreen (refactored search view)
- implemented local storage access for recently searched writing
- WritingRow (general component for any listed writing)
  - doesn't  need a parent WritingsList component
    - decided to do without WritingsList once I realized it wasn't adding any logic. I think this is both simpler and flatter for now? Tell me what you think.
    - borders are moved from WritingsList to root container of WritingRow
  - bigger touchable area
  - this could also replace RoomWritingListItem and WritingsList in StartedRoom as well. I made these changes at first, but restored them because I wasn't sure what you'd think.
- Animation (slightly different from Transition)
  - supports uncontrolled one-off animations for more imperative use
  - found this useful in SearchSubscreen, where using Transition component for each list item would call for a corresponding condition state

### CODE REVIEW

File: expo/components/buttons/UserFilterButton.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                ...styles.userFilterButton,
+            }}
+            text={text}
+        />
+    </TouchableOpacity>
+);
+
+const styles = StyleSheet.create({
+    userFilterButtonTemp: {
+        borderRadius: PARAGRAPH_FONT_SIZE * 10,
```
Comment (afkim14) [inline expo/components/buttons/UserFilterButton.tsx:39]: Ohh I like how it's relative to the font size! Learned something new :)

File: expo/utils/modules/api.ts [hunk 2]
```diff
@@ -203,6 +203,61 @@ class Api {
...
+     * @param page - page number (0<=)
+     */
+    getSearchResult(query: string, page: number): Promise<any> {
+        // const encodedQuery = encodeURIComponent(query).replace(
+        //     /[!;()*]/g,
+        //     (c) => `%${c.charCodeAt(0).toString(16)}`,
+        // );
+        // return this.transport.get(`${this.endpoint}/writings/filter?query=${encodedQuery}&page=${page}`);
+        return this.transport.get('https://hallasan-test.azurewebsites.net/api/v2/writings/filter?room_id=ee5726a5-efa6-4aed-a402-356a65bb9703&query_text=text');
+        const test = this.userId; // handles error
```
Comment (afkim14) [inline expo/utils/modules/api.ts:219]: This code doesn't run after you `return` above ^

**Comment by afkim14** [comment]
Just adding here so we don't forget: might need to transfer search view into `StartedRoom` screen.

File: expo/components/buttons/UserFilterButton.tsx (hunk 1 above)
Comment (paukparl) [reply expo/components/buttons/UserFilterButton.tsx:39]: 아 ㅋㅋㅋㅋ 그냥 엄청 큰 value 넣어야겠다 해서 저렇게 했어요

File: expo/components/buttons/UserFilterButton.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        borderColor: WRITE_POST_BG_COLOR,
+        marginRight: PARAGRAPH_FONT_SIZE * 0.5,
+        marginBottom: PARAGRAPH_FONT_SIZE * 0.5,
+        overflow: 'hidden',
+    },
+    userFilterButton: {
+        paddingHorizontal: PARAGRAPH_FONT_SIZE * 1.8,
+        paddingVertical: PARAGRAPH_FONT_SIZE * 0.6,
+    },
+    // topBarButton: {
```
[the remark this answers is not in the document]
Comment (paukparl) [reply expo/components/buttons/UserFilterButton.tsx:50]: 지웠습니당

File: expo/components/default/DefaultTextInput.tsx [hunk 4]
```diff
@@ -17,24 +17,30 @@ import {
 
 type DefaultTextInputProps = {
     refInput?: (input: any) => void;
-    text: string;
-    onTextInput: (text: string) => void;
+    text?: string;
+    onTextInput?: (text: string) => void;
```
[the remark this answers is not in the document]
Comment (paukparl) [reply expo/components/default/DefaultTextInput.tsx:21]: 그러게요 ㅋㅋㅋㅋㅋ 예전에 한거라 왜그런지 모르겠음.. 다시 돌려놓을게요

File: expo/components/hoc/WithTransition.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
+// === React Imports ===
```
[the remark this answers is not in the document]
Comment (paukparl) [reply expo/components/hoc/WithTransition.tsx:1]: 네!

File: expo/components/writing/WritingRow.tsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+import DefaultText from '../default/DefaultText';
+import {
+    BUTTON_WITH_ICON_FONT_SIZE,
+    HEADING_TWO_FONT_SIZE,
+    DEFAULT_COLOR,
+    SUBHEADING_ONE_FONT_SIZE,
+    SUBHEADING_TWO_FONT_SIZE,
+} from '../../utils/styles';
+
+type RoomWritingListItemProps = {
```
[the remark this answers is not in the document]
Comment (paukparl) [reply expo/components/writing/WritingRow.tsx:20]: 네 ㅋㅋㅋㅋ

File: expo/screens/RoomScreen.tsx [hunk 7]
```diff
@@ -47,7 +47,6 @@ class RoomScreen extends PureComponent<RoomScreenProps, RoomScreenState> {
         const { roomId } = route.params;
         api.getRoom(roomId)
             .then((roomRes) => {
-                console.log(roomRes.data);
```
[the remark this answers is not in the document]
Comment (paukparl) [reply expo/screens/RoomScreen.tsx:50]: 아 제가 쓴건줄 알고 지웠나봐요 ㅜㅋㅋㅋㅋㅋㅋ 돌려놓을게요

File: expo/utils/modules/api.ts (hunk 2 above)
Comment (paukparl) [reply expo/utils/modules/api.ts:219]: 지우는걸 깜빡했어요 ㅠㅠ

File: expo/utils/modules/api.ts [hunk 8]
```diff
@@ -203,6 +203,61 @@ class Api {
...
+     * @param query - search query
+     * @param page - page number (0<=)
+     */
+    getSearchResult(query: string, page: number): Promise<any> {
+        // const encodedQuery = encodeURIComponent(query).replace(
+        //     /[!;()*]/g,
+        //     (c) => `%${c.charCodeAt(0).toString(16)}`,
+        // );
+        // return this.transport.get(`${this.endpoint}/writings/filter?query=${encodedQuery}&page=${page}`);
+        return this.transport.get('https://hallasan-test.azurewebsites.net/api/v2/writings/filter?room_id=ee5726a5-efa6-4aed-a402-356a65bb9703&query_text=text');
```
[the remark this answers is not in the document]
Comment (paukparl) [reply expo/utils/modules/api.ts:218]: 아 네 맞아요! 아직은 search api가 hallasan-test endpoint에만 있는걸로 알고있어요

**Pull request merged**